client/src/components/Maps/AddressSetter.jsx

```diff
@@ -1,90 +1,97 @@
-import { Alert, Button, Divider, Form, Input, Menu, Space } from 'antd'
-
-const AddressSetter = ({ setSelectedLocation }) => {
+import React, { useState } from "react";
+import { OpenLocationCode } from "open-location-code";
+import { Alert, Divider, Form, Input, Menu } from "antd";
 
+const AddressSetter = ({
+  label = "Search",
+  disabled = false,
+  setSelectedLocation,
+  googleMapSearch = false,
+}) => {
   // Alerts
-  const [showAlert, setShowAlert] = useState(false)
+  const [showAlert, setShowAlert] = useState(false);
+  const [searching, setSearching] = useState(false);
 
-  const [menuResults, setMenuResults] = useState([])
-  const [selectedMenuId, setSelectedMenuId] = useState('')
+  const [menuResults, setMenuResults] = useState([]);
+  const [selectedMenuId, setSelectedMenuId] = useState("");
 
-  const onSubmit = (values) => {
-    setShowAlert(false)
+  const onSubmit = (value) => {
+    setSearching(true);
+    setShowAlert(false);
     // search for places
     const url = `https://nominatim.openstreetmap.org/search?addressdetails=1&q=${encodeURIComponent(
-      values.search)}&format=jsonv2&limit=30`
-    fetch(url).
-      then((response) => response.json()).
-      then((responses) => {
+      value
+    )}&format=jsonv2&limit=30`;
+    fetch(url)
+      .then((response) => response.json())
+      .then((responses) => {
         // If no results found, set alert
         if (responses.length === 0) {
-          setShowAlert(true)
+          setShowAlert(true);
         } else {
-          const menuResultPlaces = []
-          let counter = 0
+          const menuResultPlaces = [];
+          let counter = 0;
           responses.forEach((response) => {
-            const name = String(response.name)
+            const name = String(response.name);
             // Name + 2 for including the ',' and space after
             const address = String(response.display_name).substring(
               name.length + 2
-            )
-            const lat = response.lat
-            const lon = response.lon
+            );
+            const lat = response.lat;
+            const lon = response.lon;
             menuResultPlaces.push({
               key: counter++,
               label: String(name).trim(),
               extra: address,
               location: { lat, lon },
-              plus_code: new OpenLocationCode().encode(lat, lon)
-            })
-          })
-          setMenuResults(menuResultPlaces)
+              plus_code: new OpenLocationCode().encode(lat, lon),
+            });
+          });
+          setMenuResults(menuResultPlaces);
         }
-      }).
-      catch((error) => console.error(error))
-  }
+      })
+      .catch((error) => console.error(error))
+      .finally(() => setSearching(false));
+  };
 
   const onSubmitFailed = (errorInfo) => {
-    console.log('Failed:', errorInfo)
-    setShowAlert(true)
-  }
+    console.log("Failed:", errorInfo);
```

Comment:
can be removed in the future.

Reply:
Not up for review atm. Just placeholder texts

Reply:
Done